include/capo/pcm.hpp

```diff
@@ -56,6 +56,8 @@ class PCM::Streamer {
 	Result<void> seek(Time stamp) noexcept;
 	Time position() const noexcept;
 
+	float progress(std::size_t rindex) const noexcept;
```

Comment:
That is not very clear, why don't you call `remain` internally?

Reply:
Yeah it's confusing for sure, problem is `index` will just cause more pointless computation converting from forward to reverse indices. The main reason this is even a function is because Music also uses it to compute positions.